Lombiq.NodeJs.Extensions/scripts/markdown-to-msbuild.js

```diff
@@ -0,0 +1,128 @@
+const fs = require('fs');
+const markdownlint = require('markdownlint');
+const path = require('path');
+const process = require('process');
+const textlint = require('textlint');
+const findRecursively = require('./find-recursively');
+
+const markdownlintConfig = {
+    default: true,
+    MD013: false,
+    MD033: {
+        allowed_elements: [
+            // A special element in GitHub to indicate a keyboard key. Other Markdown formatters that don't support it
+            // will safely ignore the tags and render the content as inline text without adverse effects.
+            'kbd',
+        ],
+    },
+};
+const textLintConfig = {
+    exclude: [
+        // License files are full of legalese, which can't and shouldn't be analyzed with tools made for normal prose.
+        'License.md',
+        // The wwwwroot directory contains built and vendor assets. Any Markdown file there is not our responsibility.
+        'wwwroot',
+    ],
+    rules: [
+        'common-misspellings',
+        'doubled-spaces',
+        // "no-dead-link", // Disabled because it can't ignore relative links and can't reliably verify GitHub URLs.
+        'no-todo',
+        'no-zero-width-spaces',
+    ],
+};
+
+function mergeConfigs(baseConfiguration, rcFileName) {
+    const customOptions = fs.existsSync(rcFileName) ? JSON.parse(fs.readFileSync(rcFileName, 'utf8')) : {};
+
+    return {
+        ...baseConfiguration,
+        ...customOptions,
+    };
+}
+
+function getMarkdownPaths() {
+    let rootDirectory = process.argv.length > 2 ? process.argv[2] : '.';
+
+    // Traverse up the path until a .NET solution file (sln) is found.
+    if (rootDirectory === '_solution_') {
+        rootDirectory = path.resolve('.');
+        while (!fs.readdirSync(rootDirectory).some((name) => name.endsWith('.sln'))) {
+            rootDirectory = path.resolve(rootDirectory, '..');
+        }
+    }
+
+    return findRecursively(rootDirectory, [/\.md$/i], [/^node_modules$/, /^\.git$/, /^obj$/, /^bin$/]);
+}
+
+function handleWarning(fileName, line, column, code, message) {
+    process.stdout.write(`\r${fileName}(${line},${column}): warning ${code}: ${message}\n`);
+}
+
+function handleError(error) {
+    const code = error.code ? error.code : 'ERROR';
```

Comment:
Isn't this the equivalent of:
```suggestion
    const code = error.code || 'ERROR';
```

Reply:
Yes, but isn't this worse for readability? Also I'm always paranoid that it turns into boolean or something.

Reply:
I find the longer version ... uhm ... unnecessarily long 😉 

Would you write this in C#?
```c#
var code = error.Code != null ? error.Code : "ERROR";
```

I think you would also shorten it to:
```c#
var code = error.Code ?? "ERROR";
```

All I'm suggesting is to do the same here.

Reply:
Those are very different operators though. `??` was literally invented as a shorthand for null-coalescing. But `||` is primarily a logical operator for doing boolean things. The wrongness of using it to get a non-`Boolean` is palpable.

Also you know, JS says that an empty string is falsey, so this isn't just a null check. In C# terms this would be its equivalent:
```csharp
var code = !string.IsNullOrEmpty(error.Code) ? error.Code : "ERROR";
```

Reply:
I've been living a happy life with `||` to be honest. And of course, you're right about the empty string in JS, but if I had mentioned that, I wouldn't have had a concise example in C# to compare to.

What if `error.Code` is `undefined` though? Then your expression would return that instead of `'ERROR` 😉 I know, it's not gonna happen becuase (!) it's your own code. Still...

Reply:
> What if error.Code is undefined though?

Wrong, `undefined` is a "falsey" value.
![image](https://user-images.githubusercontent.com/4007293/180463023-b7f04da7-7446-4440-9909-abc5b520d5f2.png)

Reply:
Oh my, was I tired 😏

Reply:
I stand behind my proposition, though.

Reply:
I don't. How strongly do you feel about it? Insist, or just agree to disagree?

Reply:
I can live with your version, but I'd write it my way the next time. So, I could agree to disagree, but that doesn't help us as a team, does it? I'm open to arguments in favor of the long version - I'll gladly change my preference if I see the sense behind it.

So far, you've mentioned
1. **clarity**; and
2. **paranoia** that it turns into boolean or something. 

For clarity, I don't see a clear advantage of either style. For the second point, I find it's rather unsubstantiated. So it seems to be rather a matter of taste so far. Which is hard to argue with (or impossible). In that case, let's disagree.

Reply:
The second point is better stated this way: we are primarily C# developers so it's the natural expectation that boolean operators return a boolean result (which causes this paranoia). If you don't have to stop and ponder for a sec when seeing it, you have probably written too much JS for your own good. 😜

Reply:
That might just be true!

Let's keep your saner version then.